src/model/include/iDynTree/Model/Indeces.h

```diff
@@ -8,30 +8,10 @@
 #ifndef IDYNTREE_INDECES_H
 #define IDYNTREE_INDECES_H
 
-#include <string>
+#ifdef __DEPRECATED
```

Comment:
I think it is automatically defined by the compiler if the deprecation warnings are enabled.